### PR TITLE
Fix jumping at node edge

### PR DESCRIPTION
**Pull request by ShadowNinja**
Rebased version of #2224 + fixes.

### CODE REVIEW

**Comment by BlockMen** [comment]
> Seems to work...

Since ```standing_on_object``` is removed, have you tested whether the jittering sounds are back?

**Comment by ShadowNinja** [comment]
@BlockMen: Nope, can you check that?

**Comment by kilbith** [comment]
Tested : https://imgrush.com/m8S4PSql2qzr (video)

The fix is effective and the jittering sounds bug ain't back.

Although I'm constantly flooded on Valgrind with this warning : 

```
WARNING[Server]: collisionMoveSimple: maximum step interval exceeded, lost movement details!
```

**Comment by kilbith** [comment]
Bug spotted with this PR, the player can jump through a node placed above him : https://imgrush.com/r2AeCq8K_siH (video)

File: src/collision.cpp [hunk 1]
```diff
@@ -247,8 +247,9 @@ collisionMoveResult collisionMoveSimple(Environment *env, IGameDef *gamedef,
 
 	bool any_position_valid = false;
 
+	// The order is important here, must be y first
+	for(s16 y = max_y-1; y >= min_y; y--)
```
Comment (kilbith) [inline src/collision.cpp]: I've solved the aforementioned bug by removing the `-1`.

**Comment by BlockMen** [comment]
@ShadowNinja seems to work for me aswell. Only question left is whether this information is usefull for other usage or should get removed completely again. Both should be possible without a problem.

@kilbith your fix seems fine :+1:

**Comment by ShadowNinja** [comment]
@kilbith, @BlockMen: Good now?

**Comment by kilbith** [comment]
Seems so...

File: src/collision.cpp [hunk 2]
```diff
@@ -404,14 +405,17 @@ collisionMoveResult collisionMoveSimple(Environment *env, IGameDef *gamedef,
 		*/
 		for (u32 boxindex = 0; boxindex < cboxes.size(); boxindex++) {
 			// Ignore if already stepped up this nodebox.
```
Comment (sfan5) [inline src/collision.cpp]: you should move the comment too

**Comment by ShadowNinja** [comment]
60dc01dc258db842e229351b871d0989e3e7d62c

**Comment by PilzAdam** [comment]
What about https://github.com/minetest/minetest/issues/132#issuecomment-7168037 ?

**Pull request closed without merging**